compiler/src/main/kotlin/io/spine/protodata/Main.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.multiple
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.options.split
+import com.github.ajalt.clikt.parameters.types.file
+import com.github.ajalt.clikt.parameters.types.path
+import com.google.common.annotations.VisibleForTesting
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorRequest
+import io.spine.protodata.renderer.Renderer
+import io.spine.protodata.renderer.RendererBuilder
+import io.spine.protodata.renderer.SourceSet
+import io.spine.protodata.subscriber.Subscriber
+import io.spine.protodata.subscriber.SubscriberBuilder
+import java.io.File
+import java.net.URLClassLoader
+import java.nio.file.Path
+
+/**
+ * Launches the CLI application.
+ *
+ * When the application is done or an unhandled error occurs, exists the process.
+ */
+public fun main(args: Array<String>): Unit = Run().main(args)
+
+/**
+ * Launches the CLI application without finishing the process after execution.
+ */
+@VisibleForTesting
+internal fun launchApp(vararg argv: String) {
+    Run().parse(argv.toList())
+}
+
+/**
+ * The main CLI command which performs the ProtoData code generation tasks.
```

Comment:
Could you please describe the CLI params here?

While I realise those may not be final at this point, it would be nice to have them in order to understand the idea. It's way too long to understand them from the code.